src/components/repositories/RepositorySettings.tsx

```diff
@@ -14,76 +13,175 @@ import React from 'react';
-    this.setState(() => ({
-      needsApproval: isInputChecked,
+  handleChange = (field: keyof State) => {
+    return event => {
+      let value = event.target.value;
+      this.setState(prevState => ({
+        ...prevState,
+        [field]: value,
+      }));
+    };
+  };
+
+  changeField = (field: keyof State) => {
+    return event => {
+      let value = event.target.value;
+      this.setState(prevState => ({
+        ...prevState,
+        [field]: value,
+      }));
+    };
+  };
+
+  toggleField = (field: keyof State) => {
+    return event => {
+      this.setState(prevState => ({
+        ...prevState,
+        [field]: !prevState[field],
+      }));
+    };
+  };
+
+  addNewEnvVariable = () => {
+    this.setState(prevState => ({
+      ...prevState,
+      additionalEnvironment: (prevState.additionalEnvironment || []).concat(prevState.additionalEnvironmentToAdd),
+      additionalEnvironmentToAdd: '',
     }));
-  }
+  };
 
-  handleDecryptEnvironmentVariablesChange(event) {
-    this.setState(() => ({
-      decryptEnvironmentVariables: event.target.value,
+  deleteEnv = line => {
+    this.setState(prevState => ({
+      ...prevState,
+      additionalEnvironment: (prevState.additionalEnvironment || []).filter(value => line !== value),
     }));
-  }
+  };
 
   render() {
     let areSettingsTheSame =
       this.state.needsApproval === this.initialSettings.needsApproval &&
+      this.state.configResolutionStrategy === this.initialSettings.configResolutionStrategy &&
+      JSON.stringify(this.state.additionalEnvironment) === JSON.stringify(this.initialSettings.additionalEnvironment) &&
       this.state.decryptEnvironmentVariables === this.initialSettings.decryptEnvironmentVariables;
     return (
       <Card>
-        <CardHeader title="Security Preferences" />
         <CardContent>
           <FormControl style={{ width: '100%' }}>
             <FormControlLabel
-              control={<Switch checked={this.state.needsApproval} onChange={this.toggleNeedsApproval} />}
+              control={<Switch checked={this.state.needsApproval} onChange={this.toggleField('needsApproval')} />}
               label="Require approval for builds from users without write permissions"
             />
           </FormControl>
           <FormControl style={{ width: '100%' }}>
             <FormHelperText>Decrypt Secured Environment Variables for builds initialized by:</FormHelperText>
             <Select
               value={this.state.decryptEnvironmentVariables}
-              onChange={this.handleDecryptEnvironmentVariablesChange}
+              onChange={this.changeField('decryptEnvironmentVariables')}
               style={{ width: '100%' }}
             >
               <MenuItem value={'EVERYONE'}>Everyone</MenuItem>
               <MenuItem value={'USERS_WITH_WRITE_PERMISSIONS'}>Only users with write permissions</MenuItem>
             </Select>
           </FormControl>
```

Comment:
I feel like this could also be made into a switch.  Just a thought.

Reply:
I feel it's not descriptive since it's not a simple yes/no but two slightly different behaviors.

Reply:
That's fair.